app/src/features/surveys/view/survey-animals/TelemetryDeviceForm.tsx

```diff
@@ -0,0 +1,104 @@
+import { FormHelperText } from '@mui/material';
+import Grid from '@mui/material/Grid';
+import CustomTextField from 'components/fields/CustomTextField';
+import SingleDateField from 'components/fields/SingleDateField';
+import TelemetrySelectField from 'components/fields/TelemetrySelectField';
+import { Form, useFormikContext } from 'formik';
+import useDataLoader from 'hooks/useDataLoader';
+import { useTelemetryApi } from 'hooks/useTelemetryApi';
+import moment from 'moment';
+import { useEffect, useState } from 'react';
+import { IAnimalTelemetryDevice } from './animal';
+
+const TelemetryDeviceForm = () => {
+  const { values, setStatus } = useFormikContext<IAnimalTelemetryDevice>();
+  const [bctwErrors, setBctwErrors] = useState<Record<string, string | undefined>>({});
+  const api = useTelemetryApi();
+  const {
+    data: deviceData,
+    load: loadDevice,
+    refresh
+  } = useDataLoader(() => api.devices.getDeviceDetails(values.device_id));
+
+  useEffect(() => {
+    loadDevice();
+    refresh();
```

Comment:
`DataLoader.refresh` will call its `load` method under the hood on first load, so no need to call `loadDevice` here.